checker/check.py

```diff
@@ -0,0 +1,254 @@
+    config_yaml = yaml.load(config_text, Loader=yaml.FullLoader)
+    for git_cleanup in settings.GIT_CLEANUP_LIST:
+        config_yaml["scm"] = config_yaml["scm"].replace(git_cleanup, "")
+    return config_yaml
+
+
+def get_config_environment_names(config_yaml):
+    environments=[]
+    for environment_yaml in config_yaml["environments"]:
+        environments.append(environment_yaml["environment"])
+    return environments
+
+
+def write_csv(write_mode, columns):
+    f = open(settings.CSV_OUTPUT_FILENAME, write_mode)
+    writer = csv.writer(f)
+    writer.writerow(columns)
+    f.close()
+
+def write_headers(App, Env):
+    write_csv("w",
+        [ field.name for field in App._meta.get_fields() if not isinstance(field,models.ManyToOneRel) ] + 
+        [ field.name for field in Env._meta.get_fields() if not isinstance(field,models.ManyToOneRel) ]
+    )
+
+def write_record(app, env):
+    write_csv("a",
+        [ getattr(app,field.name) for field in app._meta.get_fields() if not isinstance(field,models.ManyToOneRel) ] +
+        [ getattr(env,field.name) for field in env._meta.get_fields() if not isinstance(field,models.ManyToOneRel) ]
+    )
+    if app:
+        app.save()
+        log.info("{} - Saved app record. ID is {}.".format(app.config_filename, app.id))
+    else:
+        log.warning("App record is not defined.")
+    if env:
+        env.save()
+        log.info("{} - Saved env record. ID is {}.".format(app.config_filename, env.id))
+    else:
+        log.warning("Env record is not defined.")
+
+
+def run_check():
+    scan_start_time=datetime.utcnow().replace(tzinfo=pytz.utc)
+    write_headers(PipelineApp, PipelineEnv)
+
+    # Initialise Github object
+    g = Github(settings.GITHUB_TOKEN)
+
+    # Initialise CloudFoundry object
+    cf_username = settings.CF_USERNAME
+    cf_password = settings.CF_PASSWORD
+    target_endpoint = settings.CF_ENDPOINT
+    cf = CloudFoundryClient(target_endpoint, proxy=dict(http=settings.CF_PROXY, https=settings.CF_PROXY))
+    cf.init_with_user_credentials(cf_username, cf_password)
+
+    # Read the pipeline configs
+    pipeline_config_repo = g.get_repo(settings.GIT_PIPELINE_REPO)
+    log.info("Config Repo: {}".format(pipeline_config_repo.name))
+    log.info("Default branch: {}".format(pipeline_config_repo.default_branch))
+    pipeline_files=get_pipeline_configs(pipeline_config_repo)
+    log.debug("Pipelines: {}".format(pipeline_files))
+
+    for pipeline_file in pipeline_files:
+        # Process pipelines
+        pipeline_app = PipelineApp()
+        pipeline_app.set_attribute("config_filename", pipeline_file)
+        pipeline_app.set_attribute("scan_start_time", scan_start_time)
+        pipeline_app.set_attribute("repo_scan_start_time", datetime.utcnow().replace(tzinfo=pytz.utc))
+
+        pipeline_app.config_filename = pipeline_file
+        log.info("START Processing pipeline file: {}".format(pipeline_app.config_filename))
```

Comment:
```suggestion
        log.info(f"START Processing pipeline file: {pipeline_app.config_filename}")
```

python f-strings make .format more readable

Reply:
Yes, that's a good shout. I'd been thinking about changing those over.
Done.